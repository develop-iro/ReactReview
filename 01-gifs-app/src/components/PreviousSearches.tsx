import type { FC } from "react";

interface Props {
  title: string;
  searches: string[];
  onSearchClick?: (search: string) => void;
}

export const PreviousSearches: FC<Props> = ({
  title,
  searches,
  onSearchClick,
}) => {
  return (
    <div className="previous-searches">
      <h2>{title}</h2>
      <ul className="previous-searches-list">
        {searches.map((search) => (
          <li key={search} onClick={() => onSearchClick?.(search)}>
            {search}
          </li>
        ))}
      </ul>
    </div>
  );
};
