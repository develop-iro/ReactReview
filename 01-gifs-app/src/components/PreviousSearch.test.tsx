import { describe, it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import { PreviousSearches } from "./PreviousSearches";

describe("PreviousSearches", () => {
  it("should render correctly", () => {
    const { container } = render(<PreviousSearches title={""} searches={[]} />);
    expect(container).toMatchSnapshot();
  });

  it("should display the title", () => {
    render(<PreviousSearches title={"My Searches"} searches={[]} />);
    expect(screen.getByText("My Searches")).toBeTruthy();
  });
  it("should display the searches", () => {
    render(
      <PreviousSearches title={"My Searches"} searches={["matrix", "neo"]} />
    );
    expect(screen.getByText("matrix")).toBeTruthy();
    expect(screen.getByText("neo")).toBeTruthy();
  });

  it("should call onSearchClick when a search is clicked", () => {
    const fn = vi.fn();
    render(
      <PreviousSearches
        title={"My Searches"}
        searches={["matrix", "neo"]}
        onSearchClick={fn}
      />
    );
    fireEvent.click(screen.getByText("matrix"));
    expect(fn).toHaveBeenCalledWith("matrix");
    fireEvent.click(screen.getByText("neo"));
    expect(fn).toHaveBeenCalledWith("neo");
  });
});
