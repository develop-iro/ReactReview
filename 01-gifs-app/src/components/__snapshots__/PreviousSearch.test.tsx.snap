// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PreviousSearches > should render correctly 1`] = `
<div>
  <div
    class="previous-searches"
  >
    <h2 />
    <ul
      class="previous-searches-list"
    />
  </div>
</div>
`;
